### PR TITLE
fix div 0 error in floormod

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Bug fixes
### PR changes

OPs 
### Describe

related issues
- https://github.com/PaddlePaddle/Paddle/issues/49919
- https://github.com/PaddlePaddle/Paddle/issues/49927
### Solution
- 参考floordiv增加除零判断和单测 https://github.com/Liyulingyue/Paddle/edit/div0_23/paddle/phi/kernels/funcs/elementwise_functor.h?pr=%2FPaddlePaddle%2FPaddle%2Fpull%2F49997#L572

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM

File: paddle/phi/kernels/funcs/elementwise_functor.h [hunk 1]
```diff
@@ -501,6 +501,7 @@ struct MinGradXYFunctor {
 template <typename T, typename Enable = void>
 struct RemainderFunctor {
   inline HOSTDEVICE T operator()(const T a, const T b) const {
+    PADDLE_ENFORCE(b != 0, DIV_ERROR_INFO);
```
Comment (luotao1) [inline paddle/phi/kernels/funcs/elementwise_functor.h:504]:
```
2023-01-22 10:21:48 0. PADDLE_ENFORCE is not recommended. Please use PADDLE_ENFORCE_EQ/NE/GT/GE/LT/LE or PADDLE_ENFORCE_NOT_NULL or PADDLE_ENFORCE_GPU_SUCCESS instead, see [ https://github.com/PaddlePaddle/Paddle/wiki/PADDLE_ENFORCE-Rewriting-Specification ] for details.
2023-01-22 10:21:48 You must have one RD (chenwhql (Recommend), luotao1 (Recommend) or lanxianghit) approval for the usage (either add or delete) of PADDLE_ENFORCE.
2023-01-22 10:21:48 PADDLE_ENFORCE(b != 0, DIV_ERROR_INFO);
2023-01-22 10:21:48 1. The error message you wrote in PADDLE_ENFORCE{_**} or PADDLE_THROW does not meet our error message writing specification. Possible errors include 1. the error message is empty / 2. the error message is too short / 3. the error type is not specified. Please read the specification [ https://github.com/PaddlePaddle/Paddle/wiki/Paddle-Error-Message-Writing-Specification ], then refine the error message. If it is a mismatch, please request chenwhql (Recommend), luotao1 or lanxianghit review and approve.
2023-01-22 10:21:48 The PADDLE_ENFORCE{_**} or PADDLE_THROW entries that do not meet the specification are as follows:
2023-01-22 10:21:48 PADDLE_ENFORCE(b != 0, DIV_ERROR_INFO); 
2023-01-22 10:21:48 There are 2 approved errors.
```
请修复下Approve流水线的问题

Comment (luotao1) [reply paddle/phi/kernels/funcs/elementwise_functor.h:504]: > 参考floordiv增加除零判断和单测 https://github.com/Liyulingyue/Paddle/edit/div0_23/paddle/phi/kernels/funcs/elementwise_functor.h?pr=%2FPaddlePaddle%2FPaddle%2Fpull%2F49997#L572

Got it.

**Comment by FlyingQianMM** [comment]
LGTM

**Review by FlyingQianMM — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
LGTM for PADDLE_ENFORCE

**Pull request merged**